packages/components/src/components/Log/Log.js

```diff
@@ -274,21 +274,30 @@ export class LogContainer extends Component {
     );
   };
 
-  getTrailerMessage = trailer => {
+  getTrailerMessage = ({ exitCode, reason }) => {
     const { forcePolling, intl } = this.props;
 
-    if (trailer && forcePolling) {
+    if (reason && forcePolling) {
       return intl.formatMessage({
         id: 'dashboard.logs.pending',
         defaultMessage: 'Final logs pending'
       });
     }
 
-    switch (trailer) {
+    switch (reason) {
       case 'Completed':
+        if (exitCode !== 0) {
+          return intl.formatMessage(
+            {
+              id: 'dashboard.pipelineRun.stepCompleted.withError',
+              defaultMessage: 'Step completed with exit code { exitCode }'
```

Comment:
```suggestion
              defaultMessage: 'Step completed with exit code {exitCode}'
```

minor issue here, there should be no spaces inside the braces. Most of the time it won't cause any issues but could become a problem in future so better to avoid it now.

Reply:
updated now, thanks